### ISSUE TITLE
"Unable to resolve module" from dependency

### ISSUE THREAD

**Issue by jeromecovington** [facebook/react-native#3874]
In this case, I am requiring `request-promise`, but when I start the react native project, I get the error:
`Unable to resolve module x. If you are sure the module is there, try restarting the packager.` 

This error is reported for many of the required dependencies of the `request` dependency of `request-promise`. I have restarted the packager as well as removing `node_modules` and reinstalling. In general, what does this error stem from and in the specific, what can be done to solve it?

**Comment by jeromecovington** [comment]
For example:
`Unable to resolve module "net". If you are sure the module is there, try restarting the packager.`

Are there known issues with dependencies requiring node core modules and breaking the packager?

**Comment by jeromecovington** [comment]
See - https://github.com/facebook/react-native/issues/1871

**Comment by zsszatmari** [comment]
+1

**Comment by skevy** [comment]
Hey everyone, can you all try:

1) upgrading to latest RN
2) `watchman watch-del-all`
2) `rm -rf $TMPDIR/react-native-packager-*`
3) restart the packager

And let me know if that fixes the issue?

**Comment by deminoth** [comment]
Checkout https://github.com/facebook/react-native/issues/4062.

**Comment by satya164** [comment]
@skevy May be we should add these steps to `react-native upgrade`?

**Comment by skevy** [comment]
@satya164 seems reasonable to me.

**Comment by satya164** [comment]
Node core modules cannot be used inside React Native as it's a completely different environment. closing this.

**Comment by abeauvois** [comment]
+1 @skevy

**Comment by viktor17492** [comment]
My development is stuck at this error. i have researched a lot on this but i found no perfect solution for this! Moreover , it is quite surprising to find an issue not resolved for the span of more than a year. 

please guys, do share if you find any solution for this.

**Comment by faogustavo** [comment]
Any solution? I'm stuck on this. :/

**Comment by dsapoetra** [comment]
I'm stuck at this too, hoping someone could give any advice/clue on this

**Comment by mpm** [comment]
I came here after having trouble importing the [a Wordpress REST API wrapper for node.js](https://github.com/WP-API/node-wpapi).
I had the same issue with `unable to resolve module url`. I took [inspiration from a similar issue](https://github.com/epoberezkin/ajv/issues/435) and solved it by running

`npm install --save url` (and subsequentially installing a package named `util` as well).

As @satya164 pointed out, this will probably not work for all missing modules, but in my case it was only `url` and `util` and works now.

Pay attention to the react error screen to find out what other modules are missing.
